src/core/thread/mle.cpp

```diff
@@ -1360,6 +1360,27 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
     VerifyOrExit(aMode != kAppendMeshLocalOnly, OT_NOOP);
     counter++;
 
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    {
+        // Cache Domain Unicast Address.
+        address = Get<DuaManager>().GetDomainUnicastAddress();
```

Comment:
One note is that here does not define a new local reference variable but reuse the `IP6::Address address` already defined in the beginning of the body, so it could be used directly in the below for loop.